### DISCUSSION TITLE
rec_r34_vd_none_bilstm_ctc.yml 恢复训练进行微调时，lr要设置为多少？怎样锁住之前正确的字符只更新错误字符？

### DISCUSSION THREAD

**Discussion by nissansz** [PaddlePaddle/PaddleOCR#14155] [Q&A]
rec_r34_vd_none_bilstm_ctc.yml 恢复训练进行微调时，lr要设置为多少？怎样锁住之前正确的字符只更新错误字符？

**Comment by UserWangZz** [comment]
学习率根据显卡个数和batchsize大小线性调整，锁住正确的字符只更新错误字符这个目前无法实现吧

**Comment by nissansz** [comment]
下面这种lr, 好像每次都是从0.0000开始自动增加，不用改？
Optimizer:
  name: AdamW
  beta1: 0.9
  beta2: 0.999
  epsilon: 1.e-8
  weight_decay: 0.05
  no_weight_decay_name: norm
  one_dim_param_no_weight_decay: True
  lr:
    name: Cosine
    learning_rate: 0.001 # 8gpus 192bs
    warmup_epoch: 5



下面这种设置，好像就是固定的，一直不变，这个在恢复训练时，应该改为0.00005或更小？

Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.0005
  regularizer:
    name: 'L2'
    factor: 0

**Reply by UserWangZz** [reply]
两种学习率变化策略，上面那个是Cosine变化，会先用5个epoch缓慢增加到设定的学习率，然后根据Cos变化逐步降低到0

**Comment by nissansz** [comment]
现在问题是，我如果只想训练1个epoch，该怎么修改lr和策略？

**Reply by UserWangZz** [reply]
第二种，学习率小一些即可

**Comment by nissansz** [comment]
原先用的第一种预热配置文件，也可以直接替换成第二种配置文件训练吗？

**Reply by UserWangZz** [reply]
可以

**Comment by nissansz** [comment]
如果设置为第二种，怎么判断已经无法再提升准确率

第一种是自动达到0，就是无法提升了？

**Reply by UserWangZz** [reply]
学习率跟准确率没啥关系

**Comment by nissansz** [comment]
怎么判断没法改善了？

**Reply by UserWangZz** [reply]
一般经验，loss变化不大，或者eval的时候精度变化不大时就差不多了

**Comment by nissansz** [comment]
NRTRLoss 是什么损失？有什么用？

**Comment by nissansz** [comment]
改成第二种后，测试了一下，原先的准确率全部清0了。
